### DISCUSSION TITLE
How to access camera?

### DISCUSSION THREAD

**Discussion by IvRogoz** [rainyl/opencv_dart#33] [Q&A] [answered]
Hello,

is there an example of how to process a camera stream?

Regards,

**Comment by rainyl** [comment, marked as answer]
## Update at 8.18.2025

For those who want to process CameraImage captured by camera plugin, now you can refer to https://github.com/rainyl/awesome-opencv_dart/tree/main/examples/camera_demo

## Previous answer

The cv.VideoCapture can open camera and take pictures, but if you want to use it in Flutter, to improve the performance, some extra works are required, i.e., you need to use [Texture](https://api.flutter.dev/flutter/widgets/Texture-class.html) to update the content, you can explore more in [camera plugin](https://github.com/flutter/packages/tree/main/packages/camera).

But if showing realtime camera video is not necessary, you can use cv.VideoCapture directly and get pictures with [cv.VideoCapture.read()](https://docs.opencv.org/4.x/d8/dfe/classcv_1_1VideoCapture.html).

if you do not want to write native code and implement Texture yourself, you can just combine camera plugin and opencv_dart, the example of this repo used to provide this, see here https://github.com/rainyl/opencv_dart/blob/d28d05eed46ba680413244d6bfb14de6af7d5ecf/example/lib/main.dart

**Reply by rainyl** [reply]
For those who want to play videos or use cameras with opencv_dart, you can also refer to https://github.com/rainyl/opencv_dart.examples/tree/main/texture_renderer for more inspirations.

**Comment by fadelfffar** [comment]
![Screenshot from 2024-12-12 04-55-55](https://github.com/user-attachments/assets/06bac315-e54f-46c2-a489-a7e2e58d6bc4)
Help please, I can't seem to run the texture_renderer example cause of uncompatibility with the file_picker packgage, I was trying all different file_picker version from 8.1.2 until 8.1.5 

> /home/fadel/.pub-cache/hosted/pub.dev/file_picker-8.1.5/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerPlugin.java:257: error: cannot find symbol
        if (registrar != null) {
            ^
  symbol:   variable registrar
  location: class FilePickerPlugin
/home/fadel/.pub-cache/hosted/pub.dev/file_picker-8.1.5/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerPlugin.java:260: error: cannot find symbol
            registrar.addActivityResultListener(this.delegate);
            ^
  symbol:   variable registrar
  location: class FilePickerPlugin
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':file_picker:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
